### ISSUE TITLE
Can't Login

### ISSUE THREAD

**Issue by dapplebeforedawn** [VitaliyRodnenko/geeknote#269]
I must be missing something really obvious...

```sh
# clone it
# cd into it

python setup.py install
# bunch of output

geeknote login
# enter my login and password
# > Sorry, incorrect login or password
```

I logged out and back into evernote's website to confirm that I have the correct username and password

```
# ~/.geeknote/error.log
2015-02-28 14:12:41,978 oauth login 219 : Incorrect login or password
2015-02-28 14:12:41,978 oauth login 222 : Target URL was not found in the response on login
```

I just have a basic evernote account, without 2-factor auth.

What am I missing here?

**Comment by macinnir** [comment]
+1 

I was very very excited to give this app a go after downloading it today, but alas ran into the same issue that @dapplebeforedawn mentioned above. After a little digging I noticed that the issue involves the lack of a `location` header with the Evernote response within oauth.py.  I haven't used Evernote's API before but I did notice a lack of Evernote's SDK which I assume does most of this type of connectivity magic that you've recreated in oauth.py.

After stdouting the httplib.HTTPResponse object, I noticed the required `location` header field is non-existent yet the HTTP status is 200 OK (along with no other obligatory ABORT/RETRY/FAIL messages). 

I turned on debug with your config, which (along with some stdouting of my own) produced the following snippet: 

```
oauth.py loadPage 100 : Request URL: www.evernote.com://Login.action?oauth_token=skaizer-5314.14BD33FFE49.68747470733A2F2F7777772E657665726E6F74652E636F6D.D236BF92749A8530A194514A841351A2 >  # cookieTestValue=1425178381578; JSESSIONID=C49075B871E5584C55BA1A760F7FA61A
>>> : Authorize...send: 'GET /Login.action?oauth_token=skaizer-5314.14BD33FFE49.68747470733A2F2F7777772E657665726E6F74652E636F6D.D236BF92749A8530A194514A841351A2 HTTP/1.1\r\nHost: www.evernote.com\r\nAccept-Encoding: identity\r\nContent-Length: 0\r\nCookie: cookieTestValue=1425178381578; JSESSIONID=C49075B871E5584C55BA1A760F7FA61A\r\n\r\n'
 >> : Authorize...reply: 'HTTP/1.1 200 OK\r\n'
header: Strict-Transport-Security: max-age=31536000
header: X-Frame-Options: SAMEORIGIN
header: Set-Cookie: shard=deleteme; Expires=Thu, 01-Jan-1970 00:00:10 GMT; Path=/
header: Set-Cookie: cookieTestValue=1425178381937; Version=1; Max-Age=378432000; Expires=Fri, 26-Feb-2027 02:53:01 GMT; Path=/
header: Content-Type: text/html;charset=utf-8
header: Content-Language: en
header: Content-Length: 6223
header: Date: Sun, 01 Mar 2015 02:53:01 GMT
header: Server: Evernote/1.0
OK
oauth.py loadPage 112 : Response : 200 > [('content-length', '6223'), ('content-language', 'en'), ('set-cookie', 'shard=deleteme; Expires=Thu, 01-Jan-1970 00:00:10 GMT; Path=/, cookieTestValue=1425178381937; Version=1; Max-Age=378432000; Expires=Fri, 26-Feb-2027 02:53:01 GMT; Path=/'), ('strict-transport-security', 'max-age=31536000'), ('server', 'Evernote/1.0'), ('date', 'Sun, 01 Mar 2015 02:53:01 GMT'), ('x-frame-options', 'SAMEORIGIN'), ('content-type', 'text/html;charset=utf-8')] 
```
Hope that helps

**Comment by bronxDakota** [comment]
I'm having the same issue -

>dakotabronx : geeknote $ geeknote login  
/usr/lib/python2.7/dist-packages/pkg_resources.py:1031: UserWarning: /home/dakotabronx/.python-eggs is writable by group/others and vulnerable to attack when used with get_resource_filename. Consider a more secure location (set with .set_extraction_path or the PYTHON_EGG_CACHE environment variable).
  warnings.warn(msg, UserWarning)
Login: bronx.dakota@gmail.com           
Password: 
Sorry, incorrect login or password


If that helps any.

**Comment by velvetomatoe** [comment]
+1 on this too. Not much to add other than I had previously run this on Mac OSX Mountain Lion with no issues. Just did a fresh install on Yosemite and trying to get it to work again and no dice. Thanks for a great program! Hope you can get it to work again soon.

**Comment by macinnir** [comment]
Might be worth noting: My issue above was on yosemite as well.

**Comment by chrox** [comment]
I also encountered this problem recently and found a way to fix it. Could someone test if #271 works for you?

**Comment by ErikBjare** [comment]
I have the same problem, on both Debian Wheezy (on an RPi) and on latest Ubuntu.
Same with the latest `master` and the 0.1 deb.

**Comment by ErikBjare** [comment]
@chrox Your solution worked, thanks.

**Comment by velvetomatoe** [comment]
@chrox Works like a charm! Thanks!

**Comment by rkrug** [comment]
Works after applying the patch - could you please merger this into the repo?

**Comment by VitaliyRodnenko** [comment]
Done! Thank you!

**Comment by kyleskrinak** [comment]
I'm running Geeknote 0.2a (7ea2255) I was successfully switching between two different evernote accounts. As of  Friday, however, while trying to switch between accounts, I can now no longer log in to _either_ account. I have verified that I can access both accounts with a web browser using these credentials. I'm running on Mac OS X 10.10.3 and Python 2.7.9. Same error log output: 

`2015-04-27 08:49:29,282 oauth login 226 : Incorrect login or password`
`2015-04-27 08:49:29,282 oauth login 229 : Target URL was not found in the response on login`

**Comment by oshev** [comment]
I can't login either. 

"Sorry, incorrect login or password"

Even though, I can login on the web. I've checked, the patch #271 is applied to the last version I downloaded.

**Comment by quydo** [comment]
I can login, the same problem, who can help me ;(

**Comment by charlesg** [comment]
+1

I have the latest which includes the code from #269 and #271 and I still can't login. I'm on a CentOS box with Python 2.7 3.

**Comment by johnerichumphries** [comment]
Having the same problem on ubuntu 14.04. Logged out but now cannot log back in (confirmed that this is not a password issue).

**Comment by constra** [comment]
+1 
same issue, on OSX 10.10.3, already used the #271 mentioned patch

**Comment by 8bitg** [comment]
+1

Just pulled the latest for this project and am experiencing the login issue.

**Comment by constra** [comment]
Hi @8bitg 
I just solved the issue using #290. This fix is not yet merged to the master, you can change it manually and install again. It worked for me, good luck!

**Comment by 8bitg** [comment]
@constra Thanks! I'll give this a try and let you know!
